docs/reference/config-params.md

```diff
@@ -5,196 +5,227 @@ title: Kev configuration reference
+Defines the [Pod Security Context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) for the kubernetes workload
 
-### Default: nil (not specified)
+### x-k8s.workload.podSecurity.runAsUser
 
-### Possible options: arbitrary numeric UID, example `1000`.
+This option sets up an appropriate User ID (`runAsUser` field) which specifies that for any Containers in the Pod, all processes will run with user ID as specified by the value.
 
-> kev.workload.pod-security-run-as-user:
+#### Default: nil (not specified)
+
+#### Possible options: arbitrary numeric UID, example `1000`.
+
+> x-k8s.workload.podSecurity.runAsUser:
 ```yaml
 version: 3.7
 services:
-  my-service:
-    labels:
-      kev.workload.pod-security-run-as-user: 1000
+  x-k8s:
+    workload:
+      podSecurity:
+        runAsUser: 1000
 ...
 ```
 
-## kev.workload.pod-security-run-as-group
+### x-k8s.workload.podSecurity.runAsGroup
 
-Defines the [Pod Security Context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) for the kubernetes workload. This option is concerned with setting up an appropriate Group ID (`runAsGroup` field) which specifies the primary group ID for all processes within any containers of the Pod. If this field is omitted (currently a default), the primary group ID of the container will be root(0). Any files created will also be owned by user with specified user ID (`runAsUser` field) and group ID (`runAsGroup` field) when runAsGroup is specified.
+This option sets up an appropriate Group ID (`runAsGroup` field) which specifies the primary group ID for all processes within any containers of the Pod. If this field is omitted (currently a default), the primary group ID of the container will be root(0). Any files created will also be owned by user with specified user ID (`runAsUser` field) and group ID (`runAsGroup` field) when runAsGroup is specified.
 
-### Default: nil (not specified)
+#### Default: nil (not specified)
 
-### Possible options: Arbitrary numeric GID. Example `1000`.
+#### Possible options: Arbitrary numeric GID. Example `2000`.
 
-> kev.workload.pod-security-run-as-group:
+> x-k8s.workload.podSecurity.runAsGroup:
 ```yaml
 version: 3.7
 services:
   my-service:
-    labels:
-      kev.workload.pod-security-run-as-group: 2000
+    workload:
+      podSecurity:
+        runAsGroup: 2000
 ...
 ```
 
-## kev.workload.pod-security-fs-group
+### x-k8s.workload.podSecurity.fsGroup
 
 Defines the [Pod Security Context](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) for the kubernetes workload. This option is concerned with setting up a supplementary group `fsGroup` field. If specified, all processes of the container are also part of this supplementary group ID. The owner for attached volumes and any files created in those volume will be Group ID as specified by the value of this configuration option.
```

Comment:
We could maybe drop the opening `Defines the [Pod Security ...]...` sentence as covered in the top block?